### DISCUSSION TITLE
Pcp_combine conf in METplus

### DISCUSSION THREAD

**Discussion by BinbinZhou-NOAA** [dtcenter/METplus#1795] [Configuration] [answered]
To  get CCPA's  24hr APCP at 12Z 20220315 (example) from past 6hr CCPA,  using pcp_combine directly as such:  
pcp_combine -sum  20220314_120000  6  20220315_120000  24  ourput.nc  -pcpdir  $CCPA_DIR 

I tried to use this example to understand important settings in  pcp_combine  conf. 
Following the use case example in the METplus user guide: 

VALID_BEG  = 2022031412
VALID_END = 2022031512
VALID_INCREMENT = 6H  

LEAD_SEQ = 24H

FCST_PCP_COMBINE_INPUT_ACCUMS = 6H
FCST_PCP_COMBINE_INPUT_NAMES = APCP
FCST_PCP_COMBINE_INPUT_LEVELS = Surface

FCST_PCP_COMBINE_OUTPUT_ACCUM = 24H 
FCST_PCP_COMBINE_OUTPUT_NAME = APCP

(1) Are these settings correct? 
(2) for FCST_PCP_COMBINE_INPUT_LEVELS = Surface,  it should be A06 or A6? 
(3) Still confused on "LEAD_SEQ=24H".  Any difference between LEAD_SEQ and FCST_PCP_COMBINE_OUTPUT_ACCUM ?  

Thanks!

**Comment by JohnHalleyGotway** [comment, marked as answer]
@BinbinZhou-NOAA I was going to take a look on Hera and test this out. But I don't actually know if/where the CCPA data lives on Hera. Do you know by any chance?

**Reply by BinbinZhou-NOAA** [reply]
John,

  I have copied all 4 ccpa files to HERA
in /scratch2/NCEPDEV/fv3-cam/noscrub/Binbin.Zhou/metplus_test/pcp_combine
The input 4 ccpa grib2 files are in ccpa24 sub-directory 
ccpa.t00z.grid3.06h.f00.grib2
ccpa.t06z.grid3.06h.f00.grib2
ccpa.t12z.grid3.06h.f00.grib2
ccpa.t18z.grid3.06h.f00.grib2

The conf file ccpa24h.conf is put. 

I tired to get 24h APCP from these 4 files.  In MET, I can use following to get the correct output 
pcp_combine 20220909_120000 06 20220910_120000 24 ccpa.t12z.grid3.24h.f00.nc -pcpdir  ./ccpa24 
which is also stored there.
But with METplus, there are some errors. 
/apps/ops/para/libs/intel/19.1.3.304/met/10.1.1/bin/pcp_combine -v 2 -sum 20220908_010200 06 20220909_010200 24 -pcpdir /lfs/h2/emc/ptmp/binbin.zhou/evs/tmpnwprd/evs_atmos_global_ens_prep.95234/ccpa24 -pcprx ccpa* -field 'name="APCP";' -name "APCP"  .....

The time "20220908_010200 06 20220909_010200 24" seems not correct.  

Binbin

**Reply by BinbinZhou-NOAA** [reply]
One data file ccpa.t18z.grid3.06h.f00.grib2 was updated. The old one has validation time error

**Reply by BinbinZhou-NOAA** [reply]
By setting 
VALID_BEG=2022091012
VALID_END=2022091012

Still got error in the log file:

 COMMAND:
/apps/ops/para/libs/intel/19.1.3.304/met/10.1.1/bin/pcp_combine -v 2 -sum 20220909_120000 06 20220910_120000 24 -pcpdir /lfs/h2/emc/ptmp/binbin.zhou/evs/tmpnwprd/evs_atmos_global_ens_prep.230887/ccpa24 -pcprx ccpa* /lfs/h2/emc/ptmp/binbin.zhou/evs/tmpnwprd/evs_atmos_global_ens_prep.230887/precip/ccpa.t12z.grid3.24h.f00.nc

OUTPUT:
DEBUG 2: Performing sum command: Init/In_Accum/Valid/Out_Accum Times = 20220909_120000/060000/20220910_120000/240000
DEBUG 2: Searching for 4 files with accumulation times of 060000 to sum to a total accumulation time of 240000.
ERROR  :
ERROR  : sum_data_files() -> cannot find a file with a valid time of 20220910_120000 and accumulation time of 060000 matching the regular expression "ccpa24"

But if I copy the pcp_combine COMMAND part in the log file to run standalone 

/apps/ops/para/libs/intel/19.1.3.304/met/10.1.1/bin/pcp_combine -v 2 -sum 20220909_120000 06 20220910_120000 24 -pcpdir /lfs/h2/emc/ptmp/binbin.zhou/evs/tmpnwprd/evs_atmos_global_ens_prep.230887/ccpa24 -pcprx ccpa* /lfs/h2/emc/ptmp/binbin.zhou/evs/tmpnwprd/evs_atmos_global_ens_prep.230887/precip/ccpa.t12z.grid3.24h.f00.nc

It can run successfully.  So very confused.

Furthermore, the error says  "cannot find a file with a valid time of 20220910_120000 and accumulation time of 060000 matching the regular expression "ccpa24" but the file ccpa.t12z.grid3.06h.f00.grib2 is validated at 2022091012 
Using degrib2 on this file can see   FIELD: APCP     Surface (18 -24 hr ) valid  18 hour after 2022090912:00:00 to 2022091012:00:00

**Reply by JohnHalleyGotway** [reply]
Binbin, I was able to get a successful run by copying over your `ccpa24.conf` file and making the following changes.
```
diff /scratch2/NCEPDEV/fv3-cam/noscrub/Binbin.Zhou/metplus_test/pcp_combine/ccpa24h.conf /home/John.H.Gotway/MET_Help/discussion_1795/ccpa24h.conf 
9c9
< VALID_END = 2020091012
---
> VALID_END = 2022091012
19,20c19,20
< OBS_PCP_COMBINE_INPUT_DIR = ccpa24
< OBS_PCP_COMBINE_INPUT_TEMPLATE = ccpa*
---
> OBS_PCP_COMBINE_INPUT_DIR = {INPUT_BASE}/ccpa24
> OBS_PCP_COMBINE_INPUT_TEMPLATE = ccpa
```
- The VALID_END must be >= VALID_BEG, as you've already found.
- I specified the full path to the input files instead of the relative path by including {INPUT_BASE} in the INPUT_DIR setting.
- The INPUT_TEMPLATE should not have a star in it. Looks like that's used to set the `-pcprx` regular expression command line option. I do realize this is confusion, but basically think of it as partial string matching. In regular expressions a `*` character means something very different than it does as a wildcard on the command line.

Please try out these changes and let me know if that fixes it.

**Reply by BinbinZhou-NOAA** [reply]
After change just one line 
OBS_PCP_COMBINE_INPUT_TEMPLATE = ccpa*
to 
 OBS_PCP_COMBINE_INPUT_TEMPLATE = ccpa

The run is successful. 

Pretty tricky!

Thanks!

Binbin
